realtime/redisPublisher.js

```diff
@@ -58,6 +58,19 @@ function prepareToPublish(inst, changedKeys, ignoreAttributes) {
 } // prepareToPublish
 
 /**
+ * Determine if an instance is needs to use the bot channel.
+ *
+ * @param  {Object} inst - Model instance to be published
+ * @returns {Boolean} - True when bot channel is expected
+ */
+function useBotChannel(inst){
+  if (rtUtils.isRoom(inst) || rtUtils.isBotAction(inst)) {
```

Comment:
Seems like jumping through a lot hoops here to check for whether the JSON object is a certain type. What about adding an optional "options" argument to the publishObject function, and just pass in whatever details you need like { isRoom: true, isBotAction: false, ... } -- that way the caller of the publishObject function is responsible for identifying it correctly, which is a lot less ambiguous than letting the receiver try to discern.